Source/VMware.PSDesiredStateConfiguration/Classes/DscConfigurationFileParser.ps1

```diff
@@ -84,9 +86,17 @@ class DscConfigurationFileParser {
             }
         }
 
-        # The ScriptBlock is invoked for all non DSC Configurations lines.
+        # The ScriptBlock is invoked for all non DSC Configurations lines with the PowerShell script file parameters if specified.
         $scriptContentAsString = $scriptContent.ToString()
-        $scriptBlock = [ScriptBlock]::Create($scriptContentAsString)
+        $scriptBlock = $null
+
+        if ($parameters.Count -gt 0) {
+            $scriptBlock = $this.CreateScriptBlockWithParameters($scriptContentAsString, $parameters)
```

Comment:
Instead of formatting the parameters as strings why don't you just call the script block with the given script params. Parse the params with AST and then pass the objects on the script block Invoke.